### ISSUE TITLE
Problem with beeswarm plot

### ISSUE THREAD

**Issue by juandavidgutier** [microsoft/EconML#612]
Hello @kbattocchi,

First, I am new in shap. I have two variables: 'Access' and 'Hesitant' in X and I can get the shap_values for my model. However, when I want to plot the shap_values with beeswarm, I see other different variables e.g. X0^3, X1^3 and others, but not my two variables of X. Why it happens?. I have shap version 0.39.0

I get this plot:

![image](https://user-images.githubusercontent.com/19823511/167253424-df99a104-f122-4230-99ce-7beb732ce3c8.png)

I'll appreciate a lot your cooperation.

Here is my dataset
[dataset_covid.csv](https://github.com/microsoft/EconML/files/8645097/dataset_covid.csv)

and here is my code:

`# importing required libraries
import os, warnings, random
import dowhy
import econml
from dowhy import CausalModel
import pandas as pd
import numpy as np
import econml
from econml.dml import DML, LinearDML, SparseLinearDML, NonParamDML, CausalForestDML
from econml.dr import DRLearner, ForestDRLearner, SparseLinearDRLearner
from econml.orf import DROrthoForest, DMLOrthoForest
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LassoCV
from econml.inference import BootstrapInference
import numpy as np, scipy.stats as st
import arviz as az
import scipy.stats as stats
from econml.metalearners import TLearner, SLearner, XLearner, DomainAdaptationLearner
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor, RandomForestClassifier, GradientBoostingRegressor, GradientBoostingClassifier
import matplotlib.pyplot as plt
from zepid.graphics import EffectMeasurePlot
from zepid.causal.causalgraph import DirectedAcyclicGraph
from joblib import Parallel, delayed
from econml.score import RScorer
from plotnine import ggplot, aes, geom_line, geom_ribbon, ggtitle, labs
import shap
from sklearn.model_selection import train_test_split
import warnings

def seed_everything(seed=123):
    random.seed(seed)
    np.random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    os.environ['TF_DETERMINISTIC_OPS'] = '1'

seed = 123
seed_everything(seed)
warnings.filterwarnings('ignore')
pd.set_option('display.float_format', lambda x: '%.2f' % x)

######
#import data
data = pd.read_csv("D:/dataset_covid.csv", encoding='latin-1') 
data = data.dropna()

#Constrained
Constrained = data[['output50', 'Constrained', 'SVI', 'Access', 'HC_Accessibility_Barriers', 'Hesitant', 'Sociodemographic_Barriers', 'PopDensity', 'broadband']] 
print(Constrained.std())

Yc = Constrained.output50.to_numpy() 
Tc = Constrained.Constrained.to_numpy()
Wc = Constrained[['SVI', 'HC_Accessibility_Barriers']].to_numpy().reshape(-1, 2)
Xc = Constrained[['Access', 'Hesitant']].to_numpy().reshape(-1, 2)

reg1 = lambda: GradientBoostingClassifier()
reg2 = lambda: GradientBoostingRegressor()

#Step 1: Modeling the causal mechanism
model_Constrained=CausalModel(
        data = Constrained,
        treatment=['Constrained'],
        outcome=['output50'],
        graph= """graph[directed 1 node[id "Constrained" label "Constrained"]
                    node[id "output50" label "output50"]
                    node[id "SVI" label "SVI"]
                    node[id "HC_Accessibility_Barriers" label "HC_Accessibility_Barriers"]
                    node[id "Access" label "Access"]
                    node[id "Hesitant" label "Hesitant"]
                    
                    edge[source "Access" target "SVI"]
                    edge[source "Access" target "HC_Accessibility_Barriers"]
                    edge[source "Access" target "Hesitant"]
                    edge[source "Access" target "output50"]
                                                          
                    edge[source "SVI" target "Constrained"]
                    edge[source "SVI" target "output50"]
                    
                    edge[source "HC_Accessibility_Barriers" target "Constrained"]
                    edge[source "HC_Accessibility_Barriers" target "output50"]
                    
                    edge[source "SVI" target "HC_Accessibility_Barriers"]
                    
                    edge[source "SVI" target "Hesitant"]
                    edge[source "HC_Accessibility_Barriers" target "Hesitant"]
                    
                    edge[source "Constrained" target "Hesitant"]
                              
                    edge[source "Constrained" target "output50"]
                    
                    edge[source "Hesitant" target "output50"]                     
                    ]"""
                    )
    

identified_estimand_Constrained = model_Constrained.identify_effect(proceed_when_unidentifiable=False)
print(identified_estimand_Constrained)

estimate_Constrained = DML(model_y=reg1(), model_t=reg2(), model_final=LassoCV(), discrete_treatment=False,
                           featurizer=PolynomialFeatures(degree=3),
                           linear_first_stages=False, cv=3, random_state=123)

estimate_Constrained.fit(Y=Yc, T=Tc, X=Xc, W=Wc, inference='bootstrap')  

#shap
shap_values = estimate_Constrained.shap_values(Xc[:200])
shap.plots.beeswarm(shap_values['Y0']['T0'])
`

**Comment by juandavidgutier** [comment]
Hello @kbattocchi,

Thanks a lot for the explanation.

Best regards
